hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -150,20 +150,7 @@ object HoodieSparkSqlWriter {
       case _ => throw new HoodieException("hoodie only support org.apache.spark.serializer.KryoSerializer as spark.serializer")
     }
     val tableType = HoodieTableType.valueOf(hoodieConfig.getString(TABLE_TYPE))
-    var operation = WriteOperationType.fromValue(hoodieConfig.getString(OPERATION))
-    // TODO clean up
-    // It does not make sense to allow upsert() operation if INSERT_DROP_DUPS is true
-    // Auto-correct the operation to "insert" if OPERATION is set to "upsert" wrongly
-    // or not set (in which case it will be set as "upsert" by parametersWithWriteDefaults()) .
-    if (hoodieConfig.getBoolean(INSERT_DROP_DUPS) &&
```

Comment:
Can we write some UT to check this change?

Reply:
Since we enable `INSERT_DROP_DUPS`, all the incoming write data will be de-duplicated in both `UPSERT` and `INSERT` operations, I find it's a bit hard to distinguish these two operations through the written data in this case.

We may get the operationType through the latest `$tbl_path/.hoodie/$timestamp.commit` file, do you think it's apporpriate? Any suggestions will be very appreciated.

Reply:
Just write a UT for the sql writer, validate the config inference.